### PR TITLE
[github] Better node_modules caching in SDK workflow

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

I noticed that the way to cache node modules recommended by GitHub is not so good because it still needs to run yarn that takes some time on resolving the dependencies in the workspace. Restoring yarn's cache and running yarn usually takes about 2 minutes.

However, instead of caching yarn cache, caching `node_modules` folder seems to be a better option for two reasons:
- on cache hit, we don't need to run yarn (one and a half minutes less)
- node_modules seems to be smaller than yarn's cache

Looks like someone else also noticed this thing: https://dev.to/mpocock1/how-to-cache-nodemodules-in-github-actions-with-yarn-24eh

# How

Replaced some steps in SDK workflow to cache `node_modules` instead of yarn's cache.
(also added some emojis and step names for a better look)

# Test Plan

CI job passes and is faster than previously by about 2 minutes

### CODE REVIEW

**Review by brentvatne — APPROVED** [review]
i believe this should work well as long as we don't add any postinstall scripts that produce side effects that we gitignore and don't cache.

would like @ide to review also though.

**Review by ide — CHANGES_REQUESTED** [review]
This restores `node_modules` in the workspace root directory but doesn't restore `node_modules` in each of the workspaces. Lots of files will be missing if we don't run `yarn install` when there is a cache hit.

I think this approach works for `tools` since it is not a workspace or a workspace root, but it's not robust for the main root directory.

**Comment by tsapeta** [comment]
Right. What if we use `**/node_modules` as a cache path? The action supports glob paths 🙂

File: .github/workflows/sdk.yml [hunk 1]
```diff
@@ -28,24 +28,30 @@ jobs:
...
+        id: node-modules-cache
         with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
-      - run: yarn install --frozen-lockfile
-      - name: Check packages
+          path: |
+            **/node_modules
```
Comment (ide) [inline .github/workflows/sdk.yml:43]:
```suggestion
            # See "workspaces" → "packages" in the root package.json for the source of truth of
            # which node_modules are affected by the root yarn.lock
            **/node_modules
```

**Review by ide — APPROVED** [review]

**Pull request merged**